Sources/PointFree/Session.swift

```diff
@@ -1,25 +1,21 @@
 import Foundation
 import HttpPipeline
+import Optics
 import Prelude
 import Tuple
 
-public struct Session: Codable {
-  public var flash: Flash?
-  public var userId: Database.User.Id?
-
-  public static let empty = Session(flash: nil, userId: nil)
-}
-
 public func writeSessionCookieMiddleware<A>(_ update: @escaping (Session) -> Session)
   -> (Conn<HeadersOpen, A>)
   -> IO<Conn<HeadersOpen, A>> {
 
     return { conn in
-      Response.Header
+      let value = update(conn.request.session)
+      guard value != conn.request.session else { return pure(conn) }
+      return Response.Header
         .setSignedCookie(
           key: pointFreeUserSession,
-          value: update(conn.request.session),
-          options: [],
+          value: value,
+          options: [.path("/")],
```

Comment:
We're writing to the session more now (it used to be just once, on login), and I found that browsers were setting different paths for different cookies. I've hard-coded the root path here to avoid multiple sessions from proliferating.